### ISSUE TITLE
Crashes when linking account then throws out error when reopening program

### ISSUE THREAD

**Issue by wreaver** [MCMi460/NSO-RPC#165]
Have tried 2 different accounts already and deleting the NSO-RPC folder on My Documents, but it always throws out the same result:
1. Link account to NSO-RPC, program crashes.
2. Reopen program, throws out error and closes until NSO-RPC folder is deleted.
3. Repeat

This is the error in question: 
Traceback (most recent call last):
  File "cli.py", line 90, in update
  File "api\__init__.py", line 201, in getSelf
KeyError: 'result'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "app.py", line 836, in <module>
  File "app.py", line 342, in selfService
  File "app.py", line 481, in changeState
  File "app.py", line 478, in changeState
  File "cli.py", line 95, in update
Exception: Cannot get session token properly

**Comment by MerakiMage** [comment]
I'm running into the exact same issues; My log.txt looks like this:

1732296935.437522: Successfully Connected to Discord.
1732296935.437522: Failure with authorization: 'result'\Friends returns {"status": 9406, "errorMessage": "Unauthorized.", "correlationId": "eac1a088-ec65e7d6"}
1732296935.437522: 'result'
1732296935.437522: Login from imink
1732296935.437522: 'result'
1732296935.437522: Cannot get session token properly
1732296988.727187: Successfully Connected to Discord.
1732296988.727187: Failure with authorization: 'result'\Friends returns {"status": 9406, "errorMessage": "Unauthorized.", "correlationId": "2e138923-e381c1ae"}
1732296988.727187: 'result'
1732296988.727187: Login from file
1732296988.727187: 'result'
1732296988.727187: Cannot get session token properly

**Comment by HotaruBlaze** [comment]
Yeah we're aware of the issue, its been reported in the discord and its being looked into.

We know what broke but a fix is currently unknown.

**Comment by HotaruBlaze** [comment]
Nintendo has recently updated Nintendo Switch Online (NSO), breaking support for all third-party tools. Apps like s3s, nxapi, NSO-RPC and others are currently not working.

Updating authentication credentials isn’t possible at the moment, and these services may remain unusable.

Theirs currently no ETA for a fix.

**Comment by Yeetov** [comment]
Sad :(

**Comment by luni-moon** [comment]
> Nintendo has recently updated Nintendo Switch Online (NSO), breaking support for all third-party tools. Apps like s3s, nxapi, NSO-RPC and others are currently not working.
> 
> Updating authentication credentials isn’t possible at the moment, and these services may remain unusable.
> 
> Theirs currently no ETA for a fix.

It may be a good idea to make a new issue or smth and pin this, as it seems a lot will be asking what's going on, as was I, but skimmed the other issues before doing so. I kinda figured they updated their API in some way.

**Comment by HotaruBlaze** [comment]
> > Nintendo has recently updated Nintendo Switch Online (NSO), breaking support for all third-party tools. Apps like s3s, nxapi, NSO-RPC and others are currently not working.
> > Updating authentication credentials isn’t possible at the moment, and these services may remain unusable.
> > Theirs currently no ETA for a fix.
> 
> It may be a good idea to make a new issue or smth and pin this, as it seems a lot will be asking what's going on, as was I, but skimmed the other issues before doing so. I kinda figured they updated their API in some way.

Once this PR is merged #169, Their will be a disclamer in the readme at the top telling people its broken.

**Comment by 1davi** [comment]
Any update on this?

**Comment by samuelthomas2774** [comment]
> Any update on this?

I'm not sure about the imink f-generation API used by NSO-RPC by default, but my f-generation API has worked with 3.0.3 since July (and since today 3.1.0). NSO-RPC can switch to this but it needs a lot of changes due to the additional requests required for request encryption.

**Comment by MCMi460** [comment]
> NSO-RPC can switch to this but it needs a lot of changes due to the additional requests required for request encryption.

If you're okay with this, then I'd love to make the changes! Thank you, @samuelthomas2774!

**Comment by samuelthomas2774** [comment]
> If you're okay with this, then I'd love to make the changes! Thank you, [@samuelthomas2774](https://github.com/samuelthomas2774)!

Yes of course it's fine! I've just updated the documentation at https://github.com/samuelthomas2774/nxapi-znca-api, let me know if you have any questions.

In addition to any changes necessary for it to work after the 3.0.1 update, I also ask if you use my API to make sure you request explicit user consent (e.g. a dialog explaining third-party API use when signing in/after updating), this is all documented there as well.

**Comment by MCMi460** [comment]
Will do—this sounds great.

Thank you again!

**Comment by R1quelmer** [comment]
> Farei isso — parece ótimo.
> 
> Muito obrigado novamente!

any updates?

**Comment by tsuki2704** [comment]
> > Farei isso — parece ótimo.
> > Muito obrigado novamente!
> 
> any updates?

so i think this project is dead?
